server/security/services/keycloak/AuthContextProvider.js

```diff
@@ -9,6 +9,18 @@ class KeycloakAuthContextProvider {
     }
     return null
   }
+
+  isAuthenticated () {
+    return this.getToken() !== null
```

Comment:
Minor thing. Contexts are created on every request. I've just realised how this code could be made simpler and more readable.

This class could be refactored such that `isAuthenticated` is a static value that gets initialised inside the constructor. e.g.

```js
this.isAuthenticated = (this.request.kauth && this.request.kauth.grant && this.request.kauth.grant.access_token)
```

Then we can have 


```js
isAuthenticated() {
	return this.isAuthenticated
}
```

Then `getToken` can get refactored such that:

```js
getToken() {
  if (this.isAuthenticated()) {
    return this.request.kauth.grant.access_token
  }
  return null
}
```

I think overall this is a bit simpler and it should be compatible with the new tests introduced in this PR. You don't have to do it but It might be worth a try :)

Reply:
@aliok feel free to take this or leave it. If these changes work without breaking the tests, it's an added bonus, but if the tests start failing, then I would scrap the idea for now 😃

Reply:
+1 it makes sense. will check this out

Reply:
done in https://github.com/aerogear/data-sync-server/pull/111/commits/0b0bce0e77aa86427f5e712e82468d55e53351ac